src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1465,6 +1475,74 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Returns a vertex of `sh` that has the same filtration value as `sh` if it exists, and `null_vertex()` otherwise.
+   *
+   * For a lower-star filtration built with `make_filtration_non_decreasing()`, this is a way to invert the process and find out which vertex had its filtration value propagated to `sh`.
+   * If several vertices have the same filtration value, the one it returns is arbitrary. */
+  Vertex_handle vertex_with_same_filtration(Simplex_handle sh) {
+    auto filt = filtration_(sh);
+    for(auto v : simplex_vertex_range(sh))
+      if(filtration_(find_vertex(v)) == filt)
+        return v;
+    return null_vertex();
+  }
+
+  /** \brief Returns an edge of `sh` that has the same filtration value as `sh` if it exists, and `null_simplex()` otherwise.
+   *
+   * For a flag-complex built with `expansion()`, this is a way to invert the process and find out which edge had its filtration value propagated to `sh`.
+   * If several edges have the same filtration value, the one it returns is arbitrary.
+   *
+   * \pre `sh` must have dimension at least 1. */
+  Simplex_handle edge_with_same_filtration(Simplex_handle sh) {
+#if 0
+    // FIXME: Only do this if dim >= 2, since we don't want to return a vertex...
+    // Test if we are lucky and the parent has the same filtration value.
+    Siblings* sib = self_siblings(sh);
+    Vertex_handle v_par = sib->parent();
+    sib = sib->oncles();
+    Simplex_handle par = sib->find(v_par);
+    if(filtration_(par) == filt) return edge_with_same_filtration(par);
+#endif
```

Comment:
Do we have to keep this dead code ?

Reply:
I'd like to keep at least a comment saying that it may be worth investigating if this type of thing helps. And it is convenient if that code is available somewhere easy to find, so that when we do decide to investigate, we don't have to reinvent it. But I agree that keeping dead code like that is not so good. I think I could open an issue and put that code there, with a comment in the code linking to the issue.